05_plot_cancer_type_results.ipynb

```diff
@@ -559,7 +560,7 @@
     "sig_genes = list(set([id_str.split('_')[0] for id_str in sig_ids]))\n",
     "\n",
     "# then get overlap with genes that are significant in stratified analysis\n",
-    "vogelstein_results_df = pd.read_csv(os.path.join(cfg.results_dir, 'vogelstein_stratified_pvals.tsv'),\n",
+    "vogelstein_results_df = pd.read_csv(os.path.join(results_dir, 'vogelstein_stratified_pvals.tsv'),\n",
```

Comment:
Great questions! These experiments are getting progressively more complicated, so it's super helpful that you're checking that all of this makes sense. See answers to your questions below:

> Your volcano plots are comparing genes that were predictive of mutation status in a single cancer A using 1) training set of a single cancer A vs shuffled dataset and 2) training set of pan-cancer including cancer A vs shuffled dataset. So these plots are showing there are are indeed a subset of genes (in orange) that seem to be relevant for predicting mutation status in cancer A

This is the right idea, but I'd change the wording a bit: "comparing genes that were predictive of mutation status" -> "comparing genes whose mutation status could be predicted from gene expression". So the genes in the volcano plot are the *labels* in our models (not to be confused with the gene features selected as being predictive by the model; we haven't analyzed these in detail yet, although that will likely be my next PR).

Once you make that change, what you said is exactly right.

> These venn diagrams are showing the predictive genes (to predict mutation status of a single cancer) found using single cancer training set (I think this is what you're calling your "Cancer type only") vs using stratified cancer training set. You also have a venn diagram showing the predictive genes found using pan-cancer vs using stratified cancer.

Again, small change: "predictive genes" -> "genes whose mutation status can be accurately predicted"

> I think your label for the venn diagram using pan-cancer data may need to be updated if my understanding is correct.

Good catch! I'll fix this.

> So is the takeaway here that there are some predictive genes that are predictive no matter what training dataset you start with (single, stratified, pan). But there are some genes that are found to be predictive using single and pan cancer data for training but not in stratified. Is this what you would've expected? I'm guessing that the overlap genes are those that have a robust contribution to any cancer mutation status. But the non-overlap ones for single cancer are those that have a specific contribution. I'm not sure how to explain the pan-cancer non-overlap ones.

Yes, this is exactly what I would have expected (and I agree with your summary of the takeaway):
* For any gene that can be predicted well in the stratified data set, we would expect that there are one or more individual cancer types where it can also be predicted well. This is what we see, there are almost no "stratified only" genes (except for JAK2 in the single cancer case, but this is probably just due to a lack of statistical power/limited data in individual cancers, I'm not too worried about it. If we saw a lot of these I'd be more concerned/confused).
* It's pretty widely reported that certain mutations can have strong driver effects in certain tissues/cancer types, and weaker or no effects in others. So, when we split by cancer type, we would expect that some genes show up as significant that weren't significant in the stratified dataset (i.e. the signal was "washed out" when we mix all cancer types together, but when we split cancer types out individually we see stronger effects).
* The pan-cancer non-overlap genes are less intuitive, but I think it's likely that these are just being driven by the strong signal in one or two cancers. I'm not surprised that we see results here, although it may be interesting to look further into these results (i.e. what genes/other covariate information do these rely on? if we remove certain cancer types, do these signals disappear?)

> What is the difference between pan-cancer vs stratified? Is it that they both have mixed cancer types but stratified controls for having uniform proportions of cancer and pan-cancer doesn't?

Yep, exactly. In the stratified experiments, the test set and training set both have uniform proportions of each cancer type. In the "pan-cancer" experiments, the test set is composed only of a single cancer type, and the training set is a mix of that cancer type and all the rest of the cancer types in TCGA (subject to our filters for label imbalance, etc).

So to put it slightly differently, the goal of the stratified experiments is to identify **genes** whose mutation status we can predict from gene expression, across a mixture cancer types. The goal of the "pan-cancer" experiments, on the other hand, is to identify individual **cancer types** where this prediction works well.